website/files/models/base.py

```diff
@@ -251,6 +251,29 @@ def get_or_create(cls, node, path):
             return cls.create(node=node, path=path)
 
     @classmethod
+    def get_file_guids(cls, materialized_path, provider, guids, node):
```

Comment:
It might be good to default `guids` to `None` then do `guids = guids or []`, so that the caller doesn't have to pass an empty list every time this method is invoked.